app/src/main/java/bg/zahov/app/ui/workout/start/StartWorkoutScreen.kt

```diff
@@ -0,0 +1,281 @@
+                style = MaterialTheme.typography.labelMedium,
+                color = MaterialTheme.colorScheme.secondary
+            )
+
+            Button(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(12.dp), colors = ButtonColors(
+                    containerColor = MaterialTheme.colorScheme.primaryContainer,
+                    contentColor = MaterialTheme.colorScheme.onSecondary,
+                    disabledContainerColor = MaterialTheme.colorScheme.onErrorContainer,
+                    disabledContentColor = MaterialTheme.colorScheme.background
+                ), onClick = onStartEmptyWorkout
+            ) {
+                Text(
+                    text = stringResource(R.string.start_empty_workout_text),
+                    style = MaterialTheme.typography.bodyLarge,
+                    maxLines = 1,
+                    overflow = TextOverflow.Ellipsis,
+                    color = MaterialTheme.colorScheme.onSecondary,
+                    fontWeight = FontWeight.Bold
+                )
+            }
+
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween
+            ) {
+                Text(
+                    modifier = Modifier.padding(top = 16.dp, start = 16.dp),
+                    text = stringResource(R.string.my_templates_text),
+                    style = MaterialTheme.typography.labelSmall,
+                    color = MaterialTheme.colorScheme.secondary
+                )
+
+                IconButton(onClick = onAddTemplateWorkout) {
+                    Icon(
+                        painter = painterResource(R.drawable.ic_plus),
+                        tint = Color.Unspecified,
+                        contentDescription = ""
+                    )
+                }
+            }
+
+            LazyColumn(Modifier.fillMaxSize()) {
+                items(items = workouts, key = { it.id }) {
+                    Workout(
+                        modifier = Modifier.animateItem(),
+                        workoutName = it.name,
+                        workoutDate = it.date,
+                        exercises = it.exercises,
+                        onWorkoutClick = { onWorkoutClick(it.id) },
+                        onWorkoutStart = { onWorkoutStart(it) },
+                        onEdit = { onEditWorkout(it.id) },
+                        onDelete = { onDeleteWorkout(it) },
+                        onDuplicate = { onDuplicateWorkout(it) })
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun Workout(
+    modifier: Modifier,
+    workoutName: String, workoutDate: String, exercises: List<String> = listOf(),
+    onWorkoutClick: () -> Unit,
+    onWorkoutStart: () -> Unit,
+    onEdit: () -> Unit,
+    onDelete: () -> Unit,
+    onDuplicate: () -> Unit
+) {
+    var isDropDownExpanded by remember {
```

Comment:
nit: you can use `rememberSavable` if you want the dropdown to not close on orientation change.

Reply:
👍